### PR TITLE
Upgrade react-router for the accredited-representative-portal sub application

### PR DESCRIPTION
**Pull request by chumpy**
The goal of this PR is to upgrade `react-router` for the `accredited-representative-portal` sub application.

Ticket: https://github.com/department-of-veterans-affairs/va.gov-team/issues/96895

The claim from react-router contributors is that both UX and DX dramatically increased with APIs that were introduced in v6.4 on [September 13, 2022](https://remix.run/blog/react-router-v6.4). The Accredited Representative Facing Team (ARF) who are building the Accredited Representative Portal (ARP) are considering using these newer APIs in our new site. Here is some literature from the creator Ryan Florence:

* Blog post - [Remixing React Router](https://remix.run/blog/remixing-react-router) - March 23, 2022
    * This is really concise and only a 5 minute read. I entirely defer to it for making the case on why one would be interested in the upgraded library version. So please take a look.
* Conference talk - [When To Fetch: Remixing React Router](https://www.youtube.com/watch?v=95B8mnhzoCM) - May 29, 2022
    * Ryan is a decently engaging presenter, but this isn’t necessary if you read the above.

### CODE REVIEW

File: src/applications/accredited-representative-portal/accreditation/21a/app-entry.jsx [hunk 1]
```diff
@@ -1,31 +1,5 @@
...
-// eslint-disable-next-line react-hooks/rules-of-hooks
-const history = useRouterHistory(createHistory)({
-  basename: manifest.rootUrl,
-});
 
-startReactApp(
-  <Provider store={store}>
-    <Router history={history} routes={routes} />
-  </Provider>,
-);
```
Comment (chumpy) [inline src/applications/accredited-representative-portal/accreditation/21a/app-entry.jsx:31]: This is just for the install. The 21a app isn't part of the pilot so simplifying the entry file for 21a for the upgrade install. We'll have to work out a replacement for the history usage to reenable.

File: src/applications/accredited-representative-portal/app-entry.jsx [hunk 2]
```diff
@@ -1,26 +1,19 @@
...
-import routes from './routes';
-import createReduxStore from './store';
-import rootReducer from './reducers';
-
 window.appName = manifest.entryName;
-const store = createReduxStore(rootReducer);
-connectFeatureToggle(store.dispatch);
 
 startReactApp(
   <Provider store={store}>
```
Comment (chumpy) [inline src/applications/accredited-representative-portal/app-entry.jsx:16]: We're still using the redux `Provider` stuff only for the feature flipper.

File: src/applications/accredited-representative-portal/store.js [hunk 3]
```diff
@@ -1,18 +1,6 @@
...
-    rootReducer,
-    compose(
-      applyMiddleware(thunk),
-      useDevTools ? window.__REDUX_DEVTOOLS_EXTENSION__() : f => f,
-    ),
-  );
-};
-
-export default createReduxStore;
+export default store;
```
Comment (chumpy) [inline src/applications/accredited-representative-portal/store.js:6]: We only need to satisfy a basic store to accommodate the feature flipper stuff

File: src/applications/accredited-representative-portal/pages/POARequestsPage.jsx [hunk 4]
```diff
@@ -1,17 +1,18 @@
 import React from 'react';
-
+import { useLoaderData } from 'react-router-dom';
 import DigitalSubmissionAlert from '../components/DigitalSubmissionAlert/DigitalSubmissionAlert';
-import POARequestsTableFetcher from '../components/POARequestsTableFetcher/POARequestsTableFetcher';
-import usePOARequests from '../hooks/usePOARequests';
+import POARequestsTable from '../components/POARequestsTable/POARequestsTable';
 
 const POARequestsPage = () => {
+  const { poaRequests } = useLoaderData();
```
Comment (chumpy) [inline src/applications/accredited-representative-portal/pages/POARequestsPage.jsx:7]: After the upgrade install, we may want to use a context to allow for updating this list from other places

**Comment by chumpy** [comment]
Looks like there's failing Cypress tests, working on those

**Review by nihil2501 — COMMENTED** [review]
Left some comments so far.

Also, I see a discussion of context that might be relevant in [`react-router`'s canonical auth w/ `RouterProvider` example](https://github.com/remix-run/react-router/tree/react-router%406.28.0/examples/auth-router-provider):
> The primary difference compared to how authentication was handled in BrowserRouter is that since RouterProvider decouples fetching from rendering, we can no longer rely on React context and/or hooks to get our user authentication status. We need access to this information outside of the React tree so we can use it in our route loader and action functions.

**Pull request closed without merging**